test/helpers/position_assertions.cc

```diff
@@ -1218,6 +1223,162 @@ string ApplyCompletionAssertion::toString() const {
+    // map of all .rb and .rbi files
+    UnorderedMap<string, string> sourceFiles;
+    // map of all .rbedited files whose version equals `this->version`
+    UnorderedMap<string, string> expectedEditedFiles;
+    for (auto filePath : FileOps::listFilesInDir(testDataPath, {".rb", ".rbi", ".rbedited"}, false, {}, {})) {
+        auto extension = FileOps::getExtension(filePath);
+        if (extension == "rbedited") {
+            auto extensionIndex = filePath.rfind(".rbedited", filePath.length());
+            auto fileVersion = string(FileOps::getExtension(filePath.substr(0, extensionIndex)));
+            if (fileVersion == this->version) {
+                expectedEditedFiles[filePath] = FileOps::read(filePath);
+            }
+        } else {
+            sourceFiles[filePath] = FileOps::read(filePath);
+        }
+    }
+
+    string sourceFilePath;
+    string expectedEditedFilePath;
+    string expectedEditedFileContents;
+    string actualEditedFileContents;
+    for (auto &renameItem : renameItems) {
+        // Get the file path from the edited document's uri so we can determine the path to the .rbedited file
+        index = renameItem->textDocument->uri.find(testDataPath, 0);
+        sourceFilePath = renameItem->textDocument->uri.substr(index, renameItem->textDocument->uri.length() - 1);
+        expectedEditedFilePath = updatedFilePath(sourceFilePath, this->version);
+
+        auto &edits = renameItem->edits;
+        expectedEditedFileContents = expectedEditedFiles[expectedEditedFilePath];
+        if (expectedEditedFileContents.empty()) {
+            ADD_FAIL_CHECK_AT(filename.c_str(), this->assertionLine + 1,
+                              fmt::format("Missing {} which should contain test file after applying code actions.",
+                                          expectedEditedFilePath));
+            return;
+        }
+
+        REQUIRE_FALSE(edits.empty());
+
+        // First, sort the edits by increasing starting location
+        fast_sort(edits, [](const auto &l, const auto &r) -> bool { return l->range->cmp(*r->range) < 0; });
+        // Apply the edits in the reverse order so that the indices don't change.
+        reverse(edits.begin(), edits.end());
+
+        actualEditedFileContents = string(sourceFiles[sourceFilePath]);
+        for (auto &edit : edits) {
+            auto file = core::File(string(sourceFilePath), string(actualEditedFileContents), core::File::Type::Normal);
+
+            actualEditedFileContents = applyEdit(actualEditedFileContents, file, *edit->range, edit->newText);
+        }
+        sourceFiles[sourceFilePath] = actualEditedFileContents;
+    }
+
+    // Compare every source file to its .rbedited of the same version if one exists
+    // There are 3 cases we're handling
+    // 1. present in bothsource and edited maps and contents match => success
+    // 2. present in original content, but not edited => fails
+    // 3. unexpected edit => fails
+    for (auto &it : sourceFiles) {
```

Comment:
You can actually do the following!

```suggestion
    for (auto &[filePath, actualEditedFileContents] : sourceFiles) {
```